dropshot/src/body.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2024 Oxide Computer Company
+
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use http_body_util::combinators::BoxBody;
+use http_body_util::BodyExt;
+use hyper::body::{Body as HttpBody, Bytes, Frame};
+
+type BoxError = Box<dyn std::error::Error + Send + Sync>;
+
+/// A body type for both requests and responses in Dropshot.
+#[derive(Debug)]
+pub struct Body {
+    inner: BoxBody<Bytes, BoxError>,
+}
+
+#[derive(Debug)]
+pub(crate) struct DataStream(Body);
+
+impl Body {
+    /// Create an empty body.
+    pub fn empty() -> Self {
+        let inner = http_body_util::Empty::new()
+            .map_err(|never| match never {})
+            .boxed();
+        Body { inner }
+    }
+
+    /// Create a full body from a specific buffer.
+    pub fn full(buf: impl Into<Bytes>) -> Self {
```

Comment:
Is `full` a term of art here? It's a little surprising to me as someone who isn't deep into this stuff.

Reply:
I think I first saw it years ago in netty, as something like `FullHttpMessage`. Since then, hyper has internally used `Full` as the variant for a complete, non-streaming body. And then when making the types separate, there is now `http_body_util::Full`.

Reply:
I can use a different name that you all like. Is `Body::with_content("hello world")` the preferred?